start-site/src/main/java/io/spring/start/site/extension/dependency/springmodulith/SpringModulithBuildCustomizer.java

```diff
@@ -0,0 +1,110 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.spring.start.site.extension.dependency.springmodulith;
+
+import io.spring.initializr.generator.buildsystem.Build;
+import io.spring.initializr.generator.buildsystem.Dependency;
+import io.spring.initializr.generator.buildsystem.Dependency.Builder;
+import io.spring.initializr.generator.buildsystem.DependencyScope;
+import io.spring.initializr.generator.spring.build.BuildCustomizer;
+
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * @author Oliver Drotbohm
+ */
+public class SpringModulithBuildCustomizer implements BuildCustomizer<Build> {
+
+	static final String GROUP_ID = "org.springframework.experimental";
+	static final String BOM_ARTIFACT_ID = "spring-modulith-bom";
+
+	static final Collection<String> OBSERVABILITY_DEPENDENCIES = List.of("datadog", "graphite", "influx", "new-relic",
+			"prometheus", "wavefront", "zipkin");
+
+	private static final String ARTIFACT_PREFIX = "spring-modulith-";
+
+	private static final Builder<?> ACTUATOR = Dependency.withCoordinates(GROUP_ID, ARTIFACT_PREFIX + "actuator")
+			.scope(DependencyScope.RUNTIME);
+
+	private static final Builder<?> OBSERVABILITY = Dependency
+			.withCoordinates(GROUP_ID, ARTIFACT_PREFIX + "observability")
+			.scope(DependencyScope.RUNTIME);
+
+	private static final Builder<?> STARTER_JDBC = Dependency.withCoordinates(GROUP_ID,
+			ARTIFACT_PREFIX + "starter-jdbc");
+
+	private static final Builder<?> STARTER_JPA = Dependency.withCoordinates(GROUP_ID, ARTIFACT_PREFIX + "starter-jpa");
+
+	private static final Builder<?> STARTER_MONGODB = Dependency.withCoordinates(GROUP_ID,
+			ARTIFACT_PREFIX + "starter-mongodb");
+
+	private static final Builder<?> TEST = Dependency.withCoordinates(GROUP_ID, ARTIFACT_PREFIX + "starter-test")
+			.scope(DependencyScope.TEST_COMPILE);
+
+	@Override
+	public void customize(Build build) {
+
+		var dependencies = build.dependencies();
+
+		// Actuator
+
+		if (dependencies.has("actuator")) {
+			dependencies.add("modulith-actuator", ACTUATOR);
+		}
+
+		// Observability
+
+		if (OBSERVABILITY_DEPENDENCIES.stream().anyMatch(dependencies::has)) {
+			dependencies.add("modulith-observability", OBSERVABILITY);
+		}
+
+		// Event publication registry support
+
+		var added = addEventPublicationRegistryBackend(build);
+
+		if (added) {
+			dependencies.remove("modulith");
+			build.boms().add("spring-modulith");
+		}
```

Comment:
I wonder if we should just keep the core starter dependency rather than swapping them.

Reply:
I originally wanted to avoid declaring dependencies that do not explicitly enable features, but the core starter actually enables the Moments (event-based abstraction of time) feature. With that thought in mind, it's reasonable to still explicitly include it. I've reworked the section accordingly.